src/overlays/actors/ovl_En_Dnq/z_en_dnq.c

```diff
@@ -22,55 +52,424 @@ const ActorInit En_Dnq_InitVars = {
+            this->unk_38C = Rand_S16Offset(20, 20);
+            this->unk_388 += 3;
+            this->unk_388 %= 5;
+        }
+        func_80A5257C(this, D_80A535DC[this->unk_388]);
+        this->skelAnime.animCurrentFrame = 0.0f;
+    }
+
+    this->unk_386 = 0;
+    return 0;
+}
+
+s32 func_80A52B68(EnDnq* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    u16 temp = globalCtx->msgCtx.unk11F04;
+
+    if ((player->stateFlags1 & 0x40) && (player->targetActor == &this->actor)) {
+        switch (temp) {
+            case 0x89B:
+                func_80A5257C(this, 18);
+                break;
+
+            case 0x89E:
+                func_80A5257C(this, 19);
+                break;
+
+            case 0x898:
+            case 0x89F:
+                func_80A5257C(this, 16);
+                break;
+
+            case 0x899:
+            case 0x89D:
+                func_80A5257C(this, 14);
+                break;
+
+            case 0x89A:
+            case 0x89C:
+                func_80A5257C(this, 15);
+                break;
+        }
+        this->unk_39C = 1;
+    } else if (this->unk_39C != 0) {
+        func_80A5257C(this, 0);
+        this->unk_39C = 0;
+        this->unk_37E = 0;
+        this->unk_386 = 0;
+    }
+
+    return 0;
+}
+
+void func_80A52C6C(EnDnq* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad;
+    Vec3f sp34 = { 0.0f, 0.0f, 110.0f };
+    Vec3f sp28;
+    Vec3f sp1C;
+
+    Math_Vec3f_Copy(&sp1C, &player->actor.world.pos);
+    Lib_Vec3f_TranslateAndRotateY(&this->actor.world.pos, this->actor.world.rot.y, &sp34, &sp28);
+    Math_Vec3f_Copy(&this->unk_370, &sp28);
+    this->actor.xzDistToPlayer = Math_Vec3f_DistXZ(&sp28, &sp1C);
+}
+
+s32* func_80A52CF8(EnDnq* this, GlobalContext* globalCtx) {
+    if (gSaveContext.weekEventReg[23] & 0x20) {
+        return &D_80A53400[14];
+    }
+
+    if (this->unk_3A4 != 0) {
+        return &D_80A53400[8];
+    }
+
+    return &D_80A53400[0];
+}
+
+s32 func_80A52D44(EnDnq* this, GlobalContext* globalCtx) {
+    s32 ret = false;
+
+    if ((this->unk_37C & 7) && func_800B84D0(&this->actor, globalCtx)) {
+        func_8013AED4(&this->unk_37C, 0, 7);
+        this->unk_380 = func_80A52CF8(this, globalCtx);
+        this->actionFunc = func_80A52FB8;
+        ret = true;
+    }
+
+    return ret;
+}
 
-#endif
+void func_80A52DC8(EnDnq* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 yaw = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
 
-extern ColliderCylinderInit D_80A53464;
-extern CollisionCheckInfoInit2 D_80A53490;
+    if (yaw < 0x3801) {
+        func_8013AED4(&this->unk_37C, 3, 7);
+    } else {
+        func_8013AED4(&this->unk_37C, 0, 7);
+    }
 
-extern UNK_TYPE D_0600EB48;
+    if (!(gSaveContext.weekEventReg[23] & 0x20)) {
+        this->unk_390 = 70.0f;
+        if (func_80114F2C(0x17) && !func_801690CC(globalCtx) && (func_80152498(&globalCtx->msgCtx) == 0) &&
```

Comment:
```suggestion
        if (func_80114F2C(ITEM_DEKU_PRINCESS) && !func_801690CC(globalCtx) && (func_80152498(&globalCtx->msgCtx) == 0) &&
```
The other use of `func_80114F2C` from `en_trt.c` seems to indicate that `0x17` is an itemId, and given this code and other context, it's likely the deku princess bottle. Could you add this enum to `z64item.h` as well?